lib/marc_to_argot/macros/shared/notes.rb

```diff
@@ -381,11 +382,39 @@ def note_related_work_label(field)
         when '544'
           labels << collect_subfield_values_by_code(field, '3').join(' ')
           labels << 'Related materials' if %w[0 1].include?(field.indicator1)
+        when '581'
+          if field.subfields.map(&:code).include?('3')
+            sf3 = collect_subfield_values_by_code(field, '3').join(' ')
+            labels << "Publications relating to #{sf3.downcase}"
+          else
+            labels << 'Related publications'
+          end
         end
 
         labels.compact.reject(&:empty?).join(': ')
       end
 
+      def note_related_work_value(field, spec, extractor)
+        value = []
+        case field_tag_or_880_linkage_tag(field)
+        when '535'
+          value = extractor.collect_subfields(field, spec).first
+        when '544'
+          value = extractor.collect_subfields(field, spec).first
+        when '580'
+          value = extractor.collect_subfields(field, spec).first
```

Comment:
More generally, since `case` is an expression that has a value, this block could be reworked as `value = case ...` with each `when` evaluating to the array you want to return.